var/spack/repos/builtin/packages/mummer/package.py

```diff
@@ -39,6 +39,19 @@ class Mummer(Package):
     patch('Makefile.patch')
     patch('scripts-Makefile.patch')
 
+    def patch(self):
+        """Fix mummerplot's use of defined on hashes (deprecated
+           since perl@5.6 made illegal in perl@5.20."""
+
+        # only fix this for versions where it is actually
+        # an error, not just deprecated.
```

Comment:
To be fair, I never tested my patch for ancient versions of Perl. I just assumed that most people would be building with a more recent version of Perl.